provider/slof.py

```diff
@@ -0,0 +1,159 @@
+import re
+import time
+import logging
+
+
+def get_boot_content(vm, start=0, start_str='SLOF',
+                     end_str='Successfully loaded',
+                     debug=False):
+    """
+    Get SLOF's content by reading the serial log.
+    :param vm: VM object.
+    :param start: int, the line number which start to read.
+    :param start_str: str, start content by string.
+    :param end_str: str, end content by string.
+    :param debug: bool, print the output of content.
+    :return: content and next line number.
+    """
+
+    def _catch_specify_content(current, string=None):
+        content = ''
+        with open(vm.serial_console_log) as f:
+            for line in f.readlines()[current:]:
+                content = content + line
+                if string in line:
+                    if debug:
+                        logging.info('Output of SLOF:\n%s' % content)
+                    return content, current + 1
+                else:
+                    current += 1
+            return None, current
+
+    current = start
+    with open(vm.serial_console_log) as f:
+        for line in f.readlines()[current:]:
+            if start_str in line:
+                if end_str:
+                    return _catch_specify_content(current, end_str)
+                else:
```

Comment:
`else:` can be omitted and unintended the next line

Reply:
Updated , please check. thanks.